src/checks/graphiteThreshold.check.js

```diff
@@ -32,25 +32,25 @@ class GraphiteThresholdCheck extends Check {
 	}
 
 	generateUrl(metric, period) {
-		const urlBase = this.graphiteBaseUrl + `format=json&from=-${period}&target=`;
-		const functionName = this.direction === 'above' ? 'maxSeries' : 'minSeries';
-		return urlBase + `${functionName}(${metric})`;
+		return this.graphiteBaseUrl + `format=json&from=-${period}&target=` + metric;
 	}
 
 	tick(){
 
 		return fetch(this.sampleUrl)
 			.then(fetchres.json)
 			.then(sample => {
-				const datapoints = sample[0].datapoints;
-
-				const invalidDatapoints = datapoints.filter(value => {
-					return this.direction === 'above' ?
-						value[0] && value[0] > this.threshold :
-						value[0] && value[0] < this.threshold;
+				const invalidDatapoints = sample.map(result => {
```

Comment:
if a single datapoint is outside the threshold, that means the check fail. so if it finds one it can stop. might save literally milliseconds if you use `some`, maybe (untested)

```
const failed = samples.some(sample => {
    return sample.datapoints.some(datapoint => {
        return this.direction ...
    })
});
```